### ISSUE TITLE
Small Change Request

### ISSUE THREAD

**Issue by shadowrylander** [stefansundin/atom-replace-notepad#2]
Hello!
My apologies if this is in the wrong place; I was wondering if there was a way to configure [the] file to be opened as a new tab in Atom rather than as a new instance of the program.
Thank you!

**Comment by stefansundin** [comment]
Hello! You are definitely in the right place for a feature request. :)

This actually seems to be the behavior for me already. New files open in the last window for me. I don't know why it's behaving differently for me. Weird.

**Comment by shadowrylander** [comment]
Thanks! 😆 
Interestingly, I have come to realize it may not actually be the packages fault, but Atom in general; would you happen to know of any way I can get files to open in a new tab instead of a new window in Atom? Since due to the number of packages I have, every new instance of Atom opens up extremely slowly, as I'm sure you know. 😃 Perhaps a change in the way the file calls Atom...? My apologies, I'm absolutely new to this. :man_shrugging:

**Comment by stefansundin** [comment]
Yes, it is possible to change how the program opens Atom. It should be possible to make this configurable in the addon settings, but for right now you might want to try to edit the source code yourself.

Because the package relies on a compiled C program to handle the registration and launching when you double click files, you have to edit some C code and compile it.

You would have to edit the file right about here: https://github.com/stefansundin/atom-replace-notepad/blob/815934d69499b577f77638405891d1e0249c2197/bin/replace-notepad.c#L125

There are instructions how to compile in the README file. It says you should use Cygwin, but now you can probably also use Bash on Windows. I will see if I can update the build instructions for that.

I don't know when I will have time to give this a try myself. Good luck, and let me know if you need more help.

**Comment by shadowrylander** [comment]
Ah, yes, I understand; I will attempt my best and let you know. Thank you very much! 😃

**Comment by stefansundin** [comment]
I made the changes required to pass arbitrary commands to atom.cmd. You have to download `replace-notepad.exe` from the master branch as I haven't released an update yet. Download it here: https://github.com/stefansundin/atom-replace-notepad/blob/master/bin/replace-notepad.exe?raw=true

Then copy that file and overwrite the one in: `%USERPROFILE%\.atom\packages\replace-notepad\bin\`

Then you have to open `regedit.exe` and edit a registry key. Navigate to:
```
HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe
```

Edit the `Debugger` value and add whatever arguments you want before the last `-`. For example, if you want to explicitly use `-a`, change it to something like this:
```
"C:\Users\ShadowRylander\.atom\packages\replace-notepad\bin\replace-notepad.exe" -a -
```

Hope that helps!

**Comment by stefansundin** [comment]
Oh, and I added Bash on Windows instructions here: https://github.com/stefansundin/atom-replace-notepad/tree/master/bin

**Comment by shadowrylander** [comment]
That was quick. Thank you so much for the help! Again, I'm a bit _too_ new at this, so... :man_shrugging: Thanks again! 😄
